tests/tablespaces/tablespace_utils.py

```diff
@@ -0,0 +1,113 @@
+import time
+import datetime as dt
+import psycopg2
+
+
+class PGNodeTS:
+    def __init__(
+        self,
+        port,
+        service_name,
+        monitor_node=None,
+    ):
+        self.port = port
+        self.service_name = service_name
+        self.monitor = monitor_node
+        if monitor_node is None:
+            self.monitor = self
+
+    def run_sql_query(self, query, *args):
+        """
+        Runs the given sql query with the given arguments in this postgres node
+        and returns the results. Returns None if there are no results to fetch.
+        """
+        result = None
+        conn = psycopg2.connect(self.connection_string())
+        conn.autocommit = True
+
+        with conn:
+            with conn.cursor() as cur:
+                cur.execute(query, args)
+                try:
+                    result = cur.fetchall()
+                except psycopg2.ProgrammingError:
+                    pass
+        # leaving contexts closes the cursor, however
+        # leaving contexts doesn't close the connection
+        conn.close()
+
+        return result
```

Comment:
Sounds like we could inherit from class PGNode here, or teach a few new tricks to it?

Reply:
I can look into it, I'd worry about surprising behavior for things that don't work for future devs, but I can take a look and maybe have "not implemented" for things that won't work?

Reply:
I've addressed all the other comments. I'm looking into a nice way to do this. The number of functions that aren't possible in this way of running (in that you need to run things from outside of the test to interact with them as they aren't run in the same container) makes it so most of the PGNode functions and DataNode functions can't be shared. 

There are subtle differences that needed to be addressed, so I haven't had any luck sharing the code yet. I tried a few approaches but they weren't great.

Reply:
I'll create a readme today for the new tests and re-use the `wait_for_state` and `run_sql_queries` with some class inheritance (i'll make new super-classes because my attempt to inherit from existing had too much cruft, IMO). :)

Reply:
Take a look at the last commit, it simplifies the tablespace_utils.py file, but add a wee bit of complexity to the main test utils. Let me know if you think it's worth it! (I expect that it is)